services/drupal/web/modules/custom/epa_core/epa_core.module

```diff
@@ -373,13 +374,45 @@ function epa_core_tokens($type, $tokens, array $data, array $options, Bubbleable
           $term_description = $token->replace($term_description, ['term' => $data['term']]);
           $replacements[$original] = $term_description;
         }
-
         break;
     }
   }
   return $replacements;
 }
 
+function epa_core_addtocal_alter(array &$overridable_items) {
+  $overridable_items['override_status'] = TRUE;
+  $node = \Drupal::routeMatch()->getParameter('node');
+  if ($node->bundle() == 'event') {
+    $cal_description = $node->get('field_add_to_cal_description')->value;
+    $default_description = $node->get('field_description')->value;
+    if (!empty($cal_description)) {
+      $description = "$cal_description" . "\n";
+    } elseif (!empty($default_description)) {
+      $description = "$default_description" . "\n";
+    }
+    $event_type = Term::load($node->get('field_event_type')->target_id)->getName();
+    if (!empty($event_type)) {
+      $event_type_full = "Event Type: $event_type" . "\n";
+    }
+    $event_host = $node->get('field_event_host')->value;
+    if (!empty($event_host)) {
+      $event_host_full = "Event host: $event_host" . "\n";
+    }
+    $links = $node->get('field_links')->getValue();
+    $links_string = "Links:" . "\n";
+    foreach ($links as $link) {
+      $text = $link['title'];
+      $url = $link['uri'];
```

Comment:
Unfortunately this isn't going to work.  The uri can hold an actual link, but can also hold a reference to another node.  When that's the case, you get something output that looks like this: `entity:node/276985`.

We need to do proper link generation...reference the field formatter for this field to get some idea of how to do that: `\Drupal\link\Plugin\Field\FieldFormatter\LinkFormatter::viewElements`

Reply:
I made some updates, there is probably a better way to go about putting the internal link together, suggestions welcome!